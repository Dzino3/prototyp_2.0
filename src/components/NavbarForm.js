import React from 'react';
import Kundenservice from './Kundenservice';
import './style/navbarStyle.css';


function NavbarForm({Logout, ShowsOrdersFalse, ShowsOrdersTrue , t}) {



  return (
    <>       
      <div className='navbarDiv'>
        <div className='navbarContainer'>
          <div className='logoImgNav'></div>
          <nav className='navbarNav'>
            <ul className='navbarUl'>
              <li className='li'><a onClick={()=>{ShowsOrdersTrue();}}>{t('navbarLabel1')}</a></li>
              <li className='li'><a onClick={()=>{ShowsOrdersFalse();}}>{t('navbarLabel2')}</a></li>
              <li className='li'><a className='impressum' href='https://www.deere.de/de/impressum/' target="_blank" rel="noopener noreferrer">{t('navbarLabel3')}</a></li>
              <li className='li'><Kundenservice t={t}></Kundenservice></li>
            </ul>
          </nav>
          <button onClick={Logout} className="logoutButton">{t('navbarLabel5')}</button>
        </div>
      </div>
    </>
  )
}

export default NavbarForm
