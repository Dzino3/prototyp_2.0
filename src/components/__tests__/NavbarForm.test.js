import NavbarForm from '../NavbarForm';
import React from "react";
import renderer from 'react-test-renderer';

// import {LogoutSucceeds} from "../../../__mocks__/logoutSucceeds.js";


// describe(`NavbarForm Rendering`, () => {

//     test('should render NavbarForm', function () {
//         let tree = renderer.create(
//         <NavbarForm Logout={LogoutSucceeds}></NavbarForm>);

//         expect(tree.toJSON()).toMatchSnapshot();
//     });
// })

describe(`Rendering NavbarForm`, () => {

    test('should render NavbarForm', function () {
        let tree = renderer.create(
        <NavbarForm></NavbarForm>);

        expect(tree.toJSON()).toMatchSnapshot();
    });
})