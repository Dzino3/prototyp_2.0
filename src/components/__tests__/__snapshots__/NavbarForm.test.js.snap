// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rendering NavbarForm should render NavbarForm 1`] = `
<div
  className="navbarDiv"
>
  <div
    className="navbarContainer"
  >
    <div
      className="logoImgNav"
    />
    <nav
      className="navbarNav"
    >
      <ul
        className="navbarUl"
      >
        <li
          className="li"
        >
          <a>
            Order
          </a>
        </li>
        <li
          className="li"
        >
          <a>
            Availability
          </a>
        </li>
        <li
          className="li"
        >
          <a>
            Order List
          </a>
        </li>
        <li
          className="li"
        >
          <a
            className="impressum"
            href="https://www.deere.de/de/impressum/"
            rel="noopener noreferrer"
            target="_blank"
          >
            Impressum
          </a>
        </li>
        <li
          className="li"
        >
          <details
            className="custom-select"
          >
            <summary
              className="radios"
            >
              <input
                checked={true}
                className="kundenInput"
                id="default"
                name="item"
                title="Customer support"
                type="radio"
              />
              <input
                className="kundenInput"
                id="item1"
                name="item"
                title="FAQ"
                type="radio"
              />
              <input
                className="kundenInput"
                id="item2"
                name="item"
                title="Service"
                type="radio"
              />
              <input
                className="kundenInput"
                id="item3"
                name="item"
                title="Newsletter"
                type="radio"
              />
              <input
                className="kundenInput"
                id="item4"
                name="item"
                title="Privacy policy"
                type="radio"
              />
            </summary>
            <ul
              className="list"
            >
              <li
                className="kundenLi"
              >
                <label
                  for="item1"
                >
                  <a
                    href="https://www.deere.de/de/ersatzteile-und-service/ersatzteile/suche-nach-teilen-h%C3%A4ufig-gestellte-fragen/"
                    target="blank"
                  >
                    FAQ
                  </a>
                  <span />
                </label>
              </li>
              <li
                className="kundenLi"
              >
                <label
                  for="item2"
                >
                  <a
                    href="https://www.deere.de/de/ersatzteile-und-service/service/"
                    target="blank"
                  >
                    Service
                  </a>
                </label>
              </li>
              <li
                className="kundenLi"
              >
                <label
                  for="item3"
                >
                  <a
                    href="https://www.deere.de/de/forms/digital-news-service/"
                    target="blank"
                  >
                    Newsletter
                  </a>
                </label>
              </li>
              <li
                className="kundenLi"
              >
                <label
                  for="item4"
                >
                  <a
                    href="https://www.deere.de/de/datenschutzerkl%C3%A4rung/"
                    target="blank"
                  >
                    Privacy and Data
                  </a>
                </label>
              </li>
            </ul>
          </details>
        </li>
      </ul>
    </nav>
    <button
      className="logoutButton"
    >
      Logout
    </button>
  </div>
</div>
`;
